index.html

```diff
@@ -88,49 +96,224 @@
+         .text("employment (%)")
+         .attr("x", width - 63)
+         .attr("y", 0)
+         .attr("font-family", "sans-serif")
+         .attr("font-size", "11px")
+         .attr("fill", "red");
+      
+      //Create Y axis
+      svg.append("g")
+        .attr("class", "axis")
+        .attr("transform", "translate(" + ypadding + ",0)")
+        .call(yAxis)
+        .append("text")
+          .text("unemployment (%)")
+          .attr("x", 0)
+          .attr("y", 0)
+          .attr("font-family", "sans-serif")
+          .attr("font-size", "11px")
+          .attr("fill", "blue");
+           
+      
 
+    //CALLING THE GRID
+    svg.append("g")         
+        .attr("class", "grid")
+        .attr("transform", "translate(0," + height + ")")
+        .call(make_x_axis()
+            .tickSize(-height, 0, 0)
+            .tickFormat("")
+        )
+
+    svg.append("g")         
+        .attr("class", "grid")
+        .call(make_y_axis()
+            .tickSize(-width, 0, 0)
+            .tickFormat("")
+        )
+    
     // HISTOGRAMS
 
-    // REGRESSION LINE
+    var formatCount = d3.format(",.0f");
+
+    //var margin = {top: 10, right: 30, bottom: 30, left: 30},
+        //width = 960 - margin.left - margin.right,
+        //height = 500 - margin.top - margin.bottom;
+
+    //Alex Answer: Yes, you do need to recreate the scale for the height of the rects in the histogram. Now, for each histogram, you are going to want a linear scale. This scale will have range and a domain, like other scales. 
+    //The domain should refer to your input data: What are the lowest and highest values that are important to the height of a histogram?
+    //The range should still be for the range of the output space, starting at 0, and going to the maximum height you would like the bars to going.
+    //Note that in the image Chris attached to the assignment, there is only one axis for each variable. So you don't have to recreate the axes.
+    
+    
+    xh = d3.scale.linear()
+        .rangeRound([0, width]);
+
+    /*
+    var bins = d3.layout.histogram()
+        .domain(xh.domain())
+        .thresholds(xh.ticks(4))
+        (data);
+    */
+
+    // Number of beans for X = 4
+    // x = min value of the data
+    // dx = distance between each bar 
+    // y = max value of the data 
+
+    //adding values ramdomly a bar is created. 
+    var bins = d3.layout.histogram()
+        .bins( [0, 0.3, 15] )(data)
+        //.thresholds(xh.ticks(4));
+        
+    yh = d3.scale.linear()
+        .domain([0, d3.max(bins, function(data) { return data.length; })])
+        .range([height, 0]);
+    
+   
+    var bar = svg.selectAll(".bar")
+        .data(bins)
+      .enter().append("g")
+        .attr("class", "bar")
+        .attr("transform", function(bin) {
+        //what is in x, dx, and y?  
+          console.log( "x: " + bin.x );
+          console.log( "dx: " + bin.dx );
+          console.log( "y: " + bin.y );
```

Comment:
You're on the right track here. But you maybe jumped too quickly to trying to add it to the SVG, while the histogram layout itself remains empty/broken. Hopefully the class demonstration was useful in this regard.